spec/features/teachers/user_can_login_spec.rb

```diff
@@ -0,0 +1,62 @@
+require "rails_helper"
+
+describe "As a teacher" do
+  before :each do
+    @teacher = create(:teacher)
+    visit welcome_path
+  end
+
+  it "I can log in successfully" do
+    create(:course,2)
+    create(:student,4)
+    @teacher.courses << Course.first
+    Student.first.courses << Course.first
+    Student.second.courses << Course.first
+    Student.third.course << Course.second
+    Student.fourth.course << Course.second
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@teacher)
+
+    expect(page).to eq("Welcome to HereToLearn")
+    fill_in "email", with: @teacher.email
+    fill_in "password", with: @teacher.password
+    click_on "Log In"
+    expect(current_path).to eq(dashboard_path)
+    expect(page).to have_content("Successfully logged in")
+    expect(page).to have_content("Logged in as #{@teacher.first_name} #{teacher.last_name}")
+
+    within "nav" do
+      expect(page).to have_link("Log Out")
+      expect(page).to have_link("Dashboard")
+    end
+
+    within ".courses-container" do
+      expect(page).to have_link("#{Course.first.name}")
+      expect(page).to have_link("#{Student.first.last_name}")
```

Comment:
Should the students names be expected `within "#course-#{Course.first.id}"` to add some specificity?

Reply:
I don't really like having css tags with interpolation, but we can talk about that as a group.

Reply:
I meant more that the student names should be expected to be associated with certain courses.

Currently, the test reads to me as "Somewhere within the courses expect to find each of these names", rather than "Expect A link to course 1. Within the container for course 1, expect to find the links to the students in Course 1"

The interpolation was more an example of how to potentially implement that testing specificity.

Reply:
Cool, I added another within statement for this test. For the dashboard test, I can add some more specificity.